### DISCUSSION TITLE
AI and machine learning in research projects  (Session 3 - Room 2)

### DISCUSSION THREAD

**Discussion by junranlei** [RSEAA/RSEAA.github.io#138] [Discussion topics for #RSEAA24]
Discussion can include (please feel free to make other suggestions too):
Challenge encountered when working on these projects (resource, GPU cost...?);
AI technology usage and applications;
Platforms, infrastructures (on-prem, cloud) and toolsets used;
Best practices and workflows.

**Comment by manodeep** [comment]
# Notes from the breakout room: AI & machine learning:

- Using AI for code-gen - you still need to understand the code. Short snippets can work okay
- Checking security - an installed app started running crypto…so unreliable potentially. 
- How to train LLMs etc? GPUs are expensive
    - virtual machines with GPU can be booked (to avoid the GPU cost) but need to be part of an org rather than individual researcher
    - much cheaper to train on supercompjuters (may be free?) rather than commercial clouds
    - random forests - so not much compute required for training but datasets are pretty big (few TBs) that needd to be fed in to get the outputs but access to institute mini super-computer is essential
- Access to supercomputing facilities is a big barrier
- best practices for machine-learning? May be domain-dependent
- reproducibility? what happens to the model after (presumably) the paper is published?
    - huggingface (but should we be using private infra for public research)
    - deploy on cloud? but for how long - project funds expire reasonably quickly
- TERN does not have any ML. Researchers exist that do th eanalysis, can directly access from NECTAR
    - One problem can be if the user has huge amoutn of data (and needs to be uploaded to the cloud). Since the TERN data are already on NECTAR, there are no costs/time associated with the data (i.e., data close to code execution)
    - how long does the data exist? Nothing is forever but TERN is NCRIS funded and has been for 20+ years. So realistically TERN should continue to exist
    - helps because TERN is a community infra
- Could excess resources be made available to the broader community?
- Difficult to know what tools are avialble? Silos...
    - "this is a tool I used inthis context and here is what worked"
- could we build a chatbot, say for Q&A sessions, using the GenAI tools? may be someone with experience?

**Comment by junranlei** [comment]
Many thanks Manodeep for the notes! Sorry I have to leave early. Some additional notes:
-Large data (TBs) is every expensive to run on the cloud (commercial/Google and AWS etc).
-GPU usage:
Researchers use work stations with GPU, local super computers (but need budget to purchase), NCI or Nectar Clouds (need to apply with project showing national significance or lower quota).
-We need best practices, workflows and processes for use cases.
-TERN link shared by Javier: https://coesra.tern.org.au/

There are quite a lot of open-source tools can be used to build RAG chatbot, a couple here: 
https://github.com/langgenius/dify
https://github.com/containers/ai-lab-recipes/tree/main/recipes/natural_language_processing/rag
Streamlit and Gradio can also be used to deploy chatbot app on Huggingface.

**Comment by NovaRouteAI** [comment]
The practical angle in "AI and machine learning in research projects (Session 3 - Room 2)" is usage attribution: the model bill only becomes actionable when it can be tied back to a user, feature, or workflow.

I would separate raw usage collection from cost presentation: first normalize tokens and provider metadata per request, then let dashboards group by user, workflow, or model tier.

I am testing a multi-model OpenAI-compatible API layer around official Chinese models, so the routing and usage-accounting parts of this are directly relevant to me. For RSEAA.github.io, do you need billing-grade accuracy here, or is operational usage visibility enough for the first version?
